qiita_pet/static/js/qiita.js

```diff
@@ -398,3 +398,24 @@ function format_biom_rows(data, row, for_study_list = true, samples = null) {
   proc_data_table += '</table>';
   return proc_data_table;
 }
+
+function generate_private_download_link(artifact_id){
+  $.ajax({
+    url: "/private_download/" + artifact_id,
+    method: 'POST',
+    success: function(response){
+      var newLink = $('<a>',{
+        text: response.url,
+        title: response.url,
+        href: response.url
+      });
+      $('#privateDownloadText').text('Link will expire in 7 days');
+      $('#privateDownloadText').append('<br/>')
+      $('#privateDownloadText').append(newLink)
+      $('#privateDownloadLink').collapse('show')
+  },
+  error: function(resp){
+    $('#downloadLinkText').text('Failed to Generate Download Link');
```

Comment:
Is there any additional information we could give the users for why a download might have failed?

Reply:
Nothing friendly I don't think.  A failure here indicates that the AJAX request to DownloadPrivateArtifactHandler.post failed out.  That could mean: Qiita is down, The user session id has expired so their authentication failed, The user is not (no longer?) able to view the chosen artifact according to check_artifact_access, various server side or network issues - no postgres server access etc.  

Basically the only thing they can possible do is refresh the page, and the only thing that might change is that they are forced to login again or their access to that artifact disappears out from under them.

Reply:
I see, this should be fine then. Thanks for the explanation.